### PR TITLE
Use a persistent volume for registry data.

### PR DESCRIPTION
**Pull request by mkantor**
If you often push the same layers from the same machine this will speed things up. It has the downside of eating up disk space (cleanup is easy though). The behavior can be disabled using a new `--no-cache` option.

### CODE REVIEW

**Comment by mkantor** [comment]
@kjrush I was snooping on forks and noticed [you had played around with this](https://github.com/kjrush/docker-pushmi-pullyu/commit/05eb37074d0dc0833c3fbdf4e74853c9bc3f99ef). Do you still use this? Got any feedback?

**Comment by msmcconnell** [comment]
I have been using the version on this branch successfully for a while now and vastly prefer the caching behavior for my use case.

**Comment by mkantor** [comment]
Thanks for the feedback @msmcconnell! Do you have any opinions on whether this should be behind a flag, and if so whether the default should be on or off?

I haven't been making heavy use of `docker-pushmi-pullyu` myself lately, but whenever I do I use this branch. I'm not sure my original concern about eating disk space is something to worry too much about—there are a lot of things in docker which behave that way and garbage collection for them is mostly manual, so experienced docker users are probably used to it already. If that reasoning seems sound, maybe I'll just merge this as-is without a flag.

**Comment by msmcconnell** [comment]
@mkantor I would suggest having a --no-cache flag so the default behavior would be to allow caching. This mirrors how the docker build command works, so I think it would be the most intuitive approach.

**Comment by mkantor** [comment]
Great suggestion @msmcconnell! I've added a `--no-cache` flag.

File: docker-pushmi-pullyu [hunk 1]
```diff
@@ -12,7 +12,8 @@ Usage:	$0 [OPTIONS] [USER@]HOST NAME[:TAG] [NAME[:TAG]...]
 Push images directly to a remote host (without a separate registry)
 
 Options:
-  -s, --ssh_opts string   Specify additional ssh arguments (e.g. --ssh_opts \"-i private.pem -C\")"
+  -s, --ssh_opts string   Specify additional ssh arguments (e.g. --ssh_opts \"-i private.pem -C\")
```
Comment (mkantor) [inline docker-pushmi-pullyu:15]: I'm planning to change the long form of this option from `--ssh_opts` to `--ssh-opts` for consistency's sake, but will do so in a separate pull request.

Comment (mkantor) [reply docker-pushmi-pullyu:15]: See #27.
